fix: normalize the case of property keys

TheLostLambda opened the pull request:
Closes #8365 and #8185

In the latest release, the case of things like `#+TITLE` were not normalized. The call `(:title ...)` would fail because the real key would be `:TITLE`.

This should finish off those two issues and I think it also cleans up the code a little.

I would like to know if there is a place to add a test for this! Lots of these org mode things get broken during development (these regressions are very common), so it would be nice to add a test or two to safeguard things from being broken again.

Let me know which file a test for parsing page titles should be in!

TheLostLambda commented:
Whoops! Left in a `print` statement! That's gone now :)

Bad3r reviewed (approved):
I used the examples provided by @emacsbliss ([**emacsbliss**](https://github.com/emacsbliss)**)**

#### The example from #8185:

filename: `abcdef.org`  
file content:

```org

:PROPERTIES:
:ID: e8f08346-452f-4e50-b8c8-a004af73cba2
:END:
#+TITLE: abc/def
```

Expected displayed page title: `abc/def`

Result: ✅ - matches expected value

#### The example from [#8365](https://github.com/logseq/logseq/issues/8365):

filename: `abcdef.org`  
file content:

```org
:PROPERTIES:
:ID:       72289d9a-eb2f-427b-ad97-b605a4b8c59b
:END:
#+title: diagram/abcdef

* manual
** anchor points
#+RESULTS:
anchor.svg
```

Expected displayed page title: `diagram/abcdef`  
Result: ✅ - matches expected value

logseq-cldwalker reviewed (commented):
Using the examples that @Bad3r gave, I can see that TITLE is fixed. The lower cased title still doesn't work for me. Having that 
@TheLostLambda Agree this cleans up the code. I'd be happy to approve once there's a unit test for this in extract_test.cljs. See https://github.com/logseq/logseq/blob/master/docs/dev-practices.md#unit-testing for working with our unit tests

TheLostLambda commented:
@logseq-cldwalker I think the problem you are having is a totally separate one! It looks like the `#+RESULTS:` line is the one causing issues with property parsing as a whole. Not sure what's going on in detail, but only the `:results` tag survives into the `properties` variable and `:title` is totally lost.

I think this should work, even though it has a lower-case `#+title:` tag :)
```org
:PROPERTIES:
:ID:       72289d9a-eb2f-427b-ad97-b605a4b8c59b
:END:
#+title: diagram/abcdef

* manual
** anchor points
anchor.svg
```

I'll add some tests and then hopefully we can close both of the linked issues (and should probably open a new one for this multiple property problem)!

TheLostLambda commented:
@logseq-cldwalker I've been trying for a little while now, but I'm actually having a bit of trouble running the tests for the `graph-parser` dependency. I've tried following the directions in the file you linked, but I don't know if that runs the tests in `deps/graph-parser/tests`?

I've tried something like `clj -M:test -d deps/graph-parser/test/ watch test`, but I'm struggling to run the right tests here! I'll try again tomorrow, but any pointers would be greatly appreciated!

logseq-cldwalker commented:
Sorry. I linked to the wrong (frontend) instructions. For the graph-parser, see https://github.com/logseq/logseq/tree/master/deps/graph-parser#testing e.g. `cd deps/graph-parser && clj -M:test`

TheLostLambda commented:
@logseq-cldwalker Finally finished things off by adding those tests! I added ones checking the casing of the property keys in particular and also added a test for checking the overall page name at the `parse-file` level so that, even if a bunch of the `graph-parser` dep is refactored, there should still be something for catching regressions!

logseq-cldwalker commented on deps/graph-parser/test/logseq/graph_parser/extract_test.cljs:
```diff
@@ -41,6 +44,28 @@
 - j"))))
 
+(deftest parse-page-title
+  []
```
This is for fn definitions and is a mistake in this file. I can clean up

logseq-cldwalker reviewed (approved):
@TheLostLambda Thanks for the fix with tests! 🐛 ⚡
